web/source/keymanweb.ts

```diff
@@ -118,12 +118,13 @@ if(!window['keyman']['initialized']) {
     util.attachDOMEvent(document, 'keyup', keymanweb.hotkeyManager._Process, false);
 
     /**
-     * Reset OSK shift states when entering or exiting the active element
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
      **/
     function resetVKShift() {
       let keyman = com.keyman.singleton;
       if(!keyman.uiManager.isActivating && keyman.osk?.vkbd) {
-        keyman.core.keyboardProcessor._UpdateVKShift(null);  //this should be enabled !!!!! TODO
+        keyman.core.resetContext(null);
```

Comment:
Added in relation to this recent comment: https://github.com/keymanapp/keyman/pull/6902#pullrequestreview-1030871819

When I thought about it for a second, this seemed like the right thing to do.  It may be a large 🔨 of sorts, but I think it makes sense to centralize the logic here, as page focus changes do seem like a cause for resetting engine context.

Reply:
I'm just a little hesitant to cherry-pick this without additional testing around it and real clarity on what the outcomes are because this is a significant change?

Reply:
There's a chance we may not actually _need_ to make a change here, but it's kind of hard to guarantee with my current understanding.  That said, I'm parsing related code paths at the moment because of the other aspect of that comment, so maybe I'll find something.

Reply:
Hmm.  For `_UpdateVKShift`, it's worth note that it's only called within `KeyboardProcessor` and triggered in two places:

1. `doModifierPress` (when a modifier is received, conditioned on `!this.contextDevice.touchable`)
2. `resetContext`.

The main KeymanEngine, however, calls _`InputProcessor`_'s version of `resetContext`, which will process the context-related keyboard groups _after_ `KeyboardProcessor.resetContext`, so we're technically fine on that angle.  At the moment.  Wouldn't hurt to add a little robustness, though.

As for the original call above... yeah, it likely did need at least _some_ sort of adjustment.

For 🍒-picking, I could change that one line to condition on `keyman.util.device.touchable` instead; that should be way less major of a change for stable.

Reply:
Okay, if that's the case, can we rename `resetVKShift` to something which clearly identifies that we're not just resetting VK shift?

Reply:
Maybe something like `updateModifierStates`?

Reply:
Ah, the joys of "end of day" brain.